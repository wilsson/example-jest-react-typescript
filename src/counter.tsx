import * as React from "react";

interface CounterState{
    count: number;
}

export class Counter extends React.Component<undefined, CounterState> {
    constructor(props) {
        super(props);
        this.state = {
            count: 0
        }
    }

    increment = (): void => {
        console.log("increment");
        this.setState({
            count: this.state.count + 1
        })
    }

    render() {
        return(
            <div>
                <h1>counter</h1>
                <button onClick = { this.increment }>click me</button>
                <div>{ this.state.count }</div>
            </div>
        )
    }
}