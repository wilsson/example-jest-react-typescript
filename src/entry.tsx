import * as React from "react";
import * as ReactDOM from "react-dom";
import { Link } from './link';
import { Counter } from './counter';

function App(){
    return(
        <div>
            <Link page="http://www.facebook.com">Facebook</Link>
            <Counter />
        </div>
    )
}
ReactDOM.render(
   <App />,
    document.getElementById("container")
);