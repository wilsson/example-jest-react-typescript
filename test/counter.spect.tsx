import * as React from 'react';
import * as Enzyme from 'enzyme';
import * as renderer from 'react-test-renderer';
import { mount } from 'enzyme';
import * as Adapter from 'enzyme-adapter-react-16';
import toJson from 'enzyme-to-json';

import { Counter } from '../src/counter';

Enzyme.configure({ adapter: new Adapter() });
it('Counter test', () => {
    let component = mount(<Counter />);
    let button = component.find('button');
    button.simulate('click');
    button.simulate('click');
    button.simulate('click');
    expect(toJson(component)).toMatchSnapshot();
});