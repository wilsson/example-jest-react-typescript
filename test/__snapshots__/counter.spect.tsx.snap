// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Counter test 1`] = `
<Counter>
  <div>
    <h1>
      counter
    </h1>
    <button
      onClick={[Function]}
    >
      click me
    </button>
    <div>
      3
    </div>
  </div>
</Counter>
`;
